runtime/stdlib/test.go

```diff
@@ -85,147 +172,107 @@ var TestContractChecker = func() *sema.Checker {
 		panic(err)
 	}
 
-	err = checker.Check()
+	err = testContractChecker.Check()
 	if err != nil {
 		panic(err)
 	}
 
-	return checker
-}()
+	variable, ok := testContractChecker.Elaboration.GetGlobalType(testContractTypeName)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+	testContractType = variable.Type.(*sema.CompositeType)
 
-func NewTestContract(
-	inter *interpreter.Interpreter,
-	testFramework TestFramework,
-	constructor interpreter.FunctionValue,
-	invocationRange ast.Range,
-) (
-	*interpreter.CompositeValue,
-	error,
-) {
-	value, err := inter.InvokeFunctionValue(
-		constructor,
-		nil,
-		testContractInitializerTypes,
-		testContractInitializerTypes,
-		invocationRange,
-	)
-	if err != nil {
-		return nil, err
+	testContractInitializerTypes = make([]sema.Type, len(testContractType.ConstructorParameters))
+	for i, parameter := range testContractType.ConstructorParameters {
+		testContractInitializerTypes[i] = parameter.TypeAnnotation.Type
 	}
 
-	compositeValue := value.(*interpreter.CompositeValue)
+	initBlockchainBackendInterfaceType()
```

Comment:
You are right, it has become unwieldy, to be honest. I will attempt to reduce most of them, as you suggested :+1:

Reply:
I have removed quite many of them, however some are not possible to be removed, since they need to be accessed in the `NewTestContract` exported function. If you have any workaround in mind, please let me know :pray: